content/master/concepts/patch-and-transform.md

```diff
@@ -0,0 +1,524 @@
+
+Connection details secret of XR is an aggregated sum of the connection details
+of the composed resources. It's recommended that the author of XRD specify
+exactly which keys will be allowed in the XR connection secret by listing them
+in `spec.connectionSecretKeys` so that consumers of claims and XRs can see what
+they can expect in the connection details secret.
+
+If `spec.connectionSecretKeys` is empty, then all keys of the aggregated connection
+details secret will be propagated.
+
+You can derive the following types of connection details from a composed
+resource to be aggregated:
+
+`FromConnectionSecretKey`. Derives an XR connection detail from a connection
+secret key of a composed resource.
+
+```yaml
+# Derive the XR's 'user' connection detail from the 'username' key of the
+# composed resource's connection secret.
+- type: FromConnectionSecretKey
+  name: user
+  fromConnectionSecretKey: username
+```
+
+`FromFieldPath`. Derives an XR connection detail from a field path within the
+composed resource.
+
+```yaml
+# Derive the XR's 'user' connection detail from the 'adminUser' status field of
+# the composed resource.
+- type: FromFieldPath
+  name: user
+  fromFieldPath: status.atProvider.adminUser
+```
+
+`FromValue`. Derives an XR connection detail from a fixed value.
+
+```yaml
+# Always sets the XR's 'user' connection detail to 'admin'.
+- type: FromValue
+  name: user
+  value: admin
+```
+
+<!-- break --> 
+
+You'll encounter a lot of 'field paths' when reading or writing a `Composition`.
+Field paths reference a field within a Kubernetes object via a simple string
+'path'. [API conventions][field-paths] describe the syntax as:
+
+> Standard JavaScript syntax for accessing that field, assuming the JSON object
+> was transformed into a JavaScript object, without the leading dot, such as
+> `metadata.name`.
+
+ Valid field paths include:
+
+* `metadata.name` - The `name` field of the `metadata` object.
+* `spec.containers[0].name` - The `name` field of the 0th `containers` element.
+* `data[.config.yml]` - The `.config.yml` field of the `data` object.
+* `apiVersion` - The `apiVersion` field of the root object.
+
+ While the following are invalid:
+
+* `.metadata.name` - Leading period.
+* `metadata..name` - Double period.
+* `metadata.name.` - Trailing period.
+* `spec.containers[]` - Empty brackets.
+* `spec.containers.[0].name` - Period before open bracket.
+
+### Patch Types
+
+You can use the following types of patch in a `Composition`:
+
+`FromCompositeFieldPath`. The default if the `type` is omitted. This type
+patches from a field within the XR to a field within the composed resource. It's
+commonly used to expose a composed resource spec field as an XR spec field.
+
+```yaml
+# Patch from the XR's spec.parameters.size field to the composed resource's
+# spec.forProvider.settings.tier field.
+- type: FromCompositeFieldPath
+  fromFieldPath: spec.parameters.size
+  toFieldPath: spec.forProvider.settings.tier
+```
```

Comment:
Note to self: I reviewed every file but this one, but only got this far down this one before I had to run.